### PR TITLE
Run golanglint-ci when a golanglint-ci config file exists

### PR DESCRIPTION
**Pull request by freeformz**
Many teams generally include some sort of linting pass in their CI configs. This is basically one tool to rule them all with a nice sentinel config file that can be used to detect intent.

cc @meekmichael

### CODE REVIEW

File: bin/test [hunk 1]
```diff
@@ -45,8 +45,12 @@ output=$(mktemp)
 case "${TOOL}" in
     gomodules)
         cd "${build}"
+        if [ -f "${build}/.golangci.yml" -o -f "${build}/.golangci.toml" -o -f "${build}/.golangci.json" ]; then
```
Comment (jmorrell) [inline bin/test:53]: Could you add an `mcount` to track usage here?

File: lib/common.sh [hunk 2]
```diff
@@ -4,6 +4,9 @@
 # load environment variables
 # allow apps to specify cgo flags. The literal text '${build_dir}' is substituted for the build directory
 
+test -n "${_COMMON_LOADED:-}" && exit 0
+export _COMMON_LOADED="true"
```
Comment (jmorrell) [inline lib/common.sh]: This is a nifty bash pattern

**Review by jmorrell — APPROVED** [review]

File: bin/test (hunk 1 above)
Comment (freeformz) [reply bin/test:53]: Sure. But, what specifically do you want to track:
* The fact that there is a config file?
* That there isn't a config file?
* That the config file is yaml? Or toml? Or json?
* That golangci-lint was run? Or not run?
* All of the above?
* Something else?

Comment (jmorrell) [reply bin/test:53]: specifically that golangci-lint was run and this feature was used

Comment (freeformz) [reply bin/test:53]: @jmorrell Reminder: this is testpack code. Does the service that execute testpack stuff support the metrics stuff that the build system supports? Last time I checked, which was admittedly a long time ago, it didn't.

Comment (jmorrell) [reply bin/test:53]: ahh, that's a very good point. Could you add a check for these files to `bin/compile` then?

**Pull request merged**